twine/auth.py

```diff
@@ -53,13 +59,41 @@ def username(self) -> Optional[str]:
     @property
     @functools.lru_cache()
     def password(self) -> Optional[str]:
+        if self.oidc:
+            # Trusted publishing (OpenID Connect): get one token from the CI
+            # system, and exchange that for a PyPI token.
+            from id import detect_credential  # type: ignore
+
+            repository_domain = cast(str, urlparse(self.system).netloc)
+            audience = self._oidc_audience(repository_domain)
+            oidc_token = detect_credential(audience)
+
+            token_exchange_url = f"https://{repository_domain}/_/oidc/mint-token"
+
+            mint_token_resp = requests.post(
```

Comment:
Using a session would be ideal. Better yet if we can share it with a `Repository` to get connection pooling and reuse for performance

Reply:
Yeah, I know the repository manages the session today. We could move that and the customization somewhere else. The repository manages it today because it was the only place to need a session. I'm not certain we need to share them, but it might be worth the effort.